services/yellow/src/contracts/Models/PaymentAuthorizationToken.cs

```diff
@@ -4,24 +4,24 @@ namespace artiso.AdsdHotel.Yellow.Contracts.Models
 {
     public class PaymentAuthorizationToken
     {
-        public PaymentAuthorizationToken()
+        public PaymentAuthorizationToken(Guid token)
         {
-            
+            Token = token.ToString();
         }
 
         public PaymentAuthorizationToken(TimeSpan duration)
         {
             Token = Guid.NewGuid().ToString();
             CreatedAt = DateTime.Now;
-            ExpirationDate = CreatedAt.Value.Add(duration);
+            ExpirationDate = CreatedAt.Add(duration);
             Active = true;
         }
 
         public string? Token { get; init; }
         
-        public DateTime? CreatedAt { get; }
+        public DateTime CreatedAt { get; }
         
-        public DateTime? ExpirationDate { get; }
+        public DateTime ExpirationDate { get; }

```

Comment:
These will have default(DateTime)  1.1.0001 if I use the GUID constructor. It this intended? I think we should make sure that all readonly properties should have a meaningful value, no matter which constructor is used.